### PR TITLE
LG-2728: migrate from Box to Polymorphic in tabs package

### PR DESCRIPTION
**Pull request by stephl3**
## ✍️ Proposed changes

- add `@leafygreen-ui/polymorphic` dependency to tabs package
- in `TabTitle` component, migrate from `Box` to `InferredPolymorphic`
- remove `@leafygreen-ui/box` dependency from tabs package

Will make final changeset after all tabs changes are merged to integration branch

🎟 _Jira ticket:_ [LG-2728](https://jira.mongodb.org/browse/LG-2728)

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/tabs/src/TabTitle/TabTitle.tsx [hunk 1]
```diff
@@ -20,89 +23,78 @@ import {
...
+    } as const;
 
-  if (typeof rest.href === 'string') {
     return (
-      <Box
-        as="a"
-        ref={titleRef as RefObject<HTMLAnchorElement>}
-        {...sharedTabProps}
-      >
+      <Component {...componentProps}>
```
Comment (TheSonOfThomp) [inline packages/tabs/src/TabTitle/TabTitle.tsx:90]: technically i believe this now allows `Tab` to render "as" any other element, not just `a` or `button`. Should add this to the changeset (minor)

File: packages/tabs/src/TabTitle/TabTitle.types.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
+import { PolymorphicAs } from '@leafygreen-ui/polymorphic';
+
 export interface BaseTabTitleProps {
+  as?: PolymorphicAs;
```
Comment (TheSonOfThomp) [inline packages/tabs/src/TabTitle/TabTitle.types.ts:4]: Can we add a control for this prop in Stortybook?

File: packages/tabs/src/Tabs/Tabs.tsx [hunk 3]
```diff
@@ -48,11 +49,11 @@ import { AccessibleTabsProps } from './Tabs.types';
  * @param props.selected Index of the Tab that should appear active. If value passed, component will be controlled by consumer.
  * @param props.setSelected Callback to be executed when Tab is selected. Receives index of activated Tab as the first argument.
  */
-const Tabs = (props: AccessibleTabsProps) => {
+const Tabs = Polymorphic<AccessibleTabsProps>(props => {
```
Comment (TheSonOfThomp) [inline packages/tabs/src/Tabs/Tabs.tsx:52]: is `Tabs` itself polymorphic? IIRC the wrapper should always be the same type of element, and only the `TabTitle` can be different

File: packages/tabs/src/TabTitle/TabTitle.types.ts (hunk 2 above)
Comment (TheSonOfThomp) [inline packages/tabs/src/TabTitle/TabTitle.types.ts:4]: Do we need to explicitly declare this prop? I believe it gets automatically added by `InferredPolymorphic`

Comment (stephl3) [reply packages/tabs/src/TabTitle/TabTitle.types.ts]: oops yeah this is a remnant from earlier

File: packages/tabs/src/Tabs/Tabs.tsx (hunk 3 above)
Comment (stephl3) [reply packages/tabs/src/Tabs/Tabs.tsx]: updated!

**Review by TheSonOfThomp — APPROVED** [review]
👍 Just needs a changeset

**Pull request merged**